typhos/display.py

```diff
@@ -457,6 +470,228 @@ def mousePressEvent(self, event):
+        """Update the tooltip based on device information."""
+        tooltip = []
+        # BUG: I'm seeing two devices in `self.devices` for
+        # $ typhos --fake-device 'ophyd.EpicsMotor[{"prefix":"b"}]'
+        for device in sorted(
+            set(self.devices),
+            key=lambda dev: self.devices.index(dev)
+        ):
+            heading = device.name or type(device).__name__
+            tooltip.extend([
+                heading,
+                "-" * len(heading),
+                ""
+            ])
+
+            tooltip.append(
+                inspect.getdoc(device) or
+                inspect.getdoc(type(device)) or
+                "No docstring"
+            )
+            tooltip.append("")
+
+        return "\n".join(tooltip)
+
+    def add_device(self, device):
+        self.devices.append(device)
+
+        self._tooltip = self._get_tooltip()
+        self.tooltip_updated.emit(self._tooltip)
+
+    @property
+    def help_url(self):
+        """The full help URL, generated from ``TYPHOS_HELP_URL``."""
+        if not self.devices:
+            return QtCore.QUrl("about:blank")
+
+        device, *_ = self.devices
+        try:
+            device_url = utils.HELP_URL.format(device=device)
+        except Exception:
+            logger.exception("Failed to format confluence URL for device %s",
+                             device)
+            return QtCore.QUrl("about:blank")
+
+        return QtCore.QUrl(device_url)
+
+    def show_help(self):
+        """Show the help information in a QWebEngineView."""
+        if self.help_web_view:
+            self.help_web_view.show()
+            return
+        self.help_web_view = QWebEngineView()
+        if QWebEngineHttpRequest is not None:
+            request = QWebEngineHttpRequest()
+            request.setUrl(self.help_url)
+            for header, value in utils.HELP_HEADERS.items():
+                request.setHeader(
+                    header.encode("utf-8"),
+                    value.encode("utf-8"),
+                )
+
+            self.help_web_view.load(request)
+        else:
+            self.help_web_view.setUrl(self.help_url)
+
+        self.help_web_view.setEnabled(True)
+        self.help_web_view.setMinimumSize(QtCore.QSize(100, 400))
+
+        self.layout().addWidget(self.help_web_view)
+
+    def hide_help(self):
+        """Hide the help information QWebEngineView."""
+        if not self.help_web_view:
+            return
+        self.help_web_view.hide()
+        if self._delete_timer is None:
+            self._delete_timer = QtCore.QTimer()
+            self._delete_timer.setInterval(20000)
+            self._delete_timer.setSingleShot(True)
+            self._delete_timer.timeout.connect(self._delete_help_if_hidden)
+            self._delete_timer.start()
+
+    def _delete_help_if_hidden(self):
+        """
+        Slowly react to the help display removal, as setting it back up can be
+        slow and painful.
```

Comment:
Does this cause some sort of problem if it isn't deleted? If it has a high startup cost, we could consider just not deleting it until the screen is closed.

Reply:
Just that it could take up an exorbitant amount of memory (imagine a browser tab for every typhos device in lucid). I thought this was a reasonable "clean up if it seems like the user isn't going to click show-help again in a few seconds".

I didn't investigate too much, though, so it's more based on a hunch.

Reply:
Gotcha, I agree that browser tabs can be memory hogs